doc/debug-system.md

```diff
@@ -0,0 +1,393 @@
+
+**Signal**               | **Direction** | **Description**
+------------------------ | ------------- | ---------------------------------------
+debug_req_o[NrHarts-1:0] | output        | Level-sensitive debug request interrupt
+
+The Debug Module issues the debug request to request the core to enter its debug mode.
+In consequence, the core is expected to
+
+- jump to the halt address in the Debug ROM,
+- save pc in dpc,
+- update dcsr.
+
+#### Auxiliary Signalling
+
+**Signal**               | **Direction** | **Description**
+------------------------ | ------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
+unavailable_i[NrHarts-1] | input         | Set to 0 to mark the hart has unavailable (e.g.: power down). This information is made available to the debugger through the dmstatus.allunavail and dmstatus.anyavail fields.
+
+#### Debug Control and Status Register (CSRs)
+
+Four debug-specific CSRs must be supported by the core, as described in the Debug Specification Section 4.8.
+
+**Address** | **Name**  | **Access**                    | **Description**
+----------- | --------- | ----------------------------- | ------------------------
+0x7b0       | dcsr      | _field-dependent; check spec_ | Debug Control and Status
+0x7b1       | dpc       | RW from Debug Mode only       | Debug PC
+0x7b2       | dscratch0 | RW from Debug Mode only       | Debug Scratch Register 0
+0x7b3       | dscratch1 | RW from Debug Mode only       | Debug Scratch Register 1
+
+#### DRET Instruction
+
+To return from debug mode the DRET instruction must be supported by the core according to the Debug Specification Section 4.7.
+
+### JTAG
+
+The JTAG Debug Transport Module (JTAG DTM) provides a standard IEEE 1149.1 JTAG connection.
+
+**Signal** | **Direction** | **Description**
+---------- | ------------- | -----------------------------------------------------------------------------------------------------
+tck_i      | input         | Test Clock
+tms_i      | input         | Test Mode Select
+td_i       | input         | Test Data In (host to debug system)
+td_o       | output        | Test Data Out (debug system to host)
+tdo_oe_o   | output        | TDO output enable
+trst_ni    | input         | Test Reset (active low). Usage of TRST is optional, but recommended for reliable reset functionality.
+
+### Bus Interface
+The Debug Module connects to the system bus as device, exposing the debug memory (the Program Buffer and the Debug ROM), and as host for the System Bus Access (SBA) functionality.
+The same generic bus interface is used in both cases.
+The bus width is configurable to be 32 or 64 bit using the `BusWidth` parameter.
+
+#### Host (Master) Interface
+
+The bus host interface is compatible to the [instruction and data interfaces of Ibex](https://ibex-core.readthedocs.io/en/latest/load_store_unit.html).
+
```

Comment:
We have internally produced some documents about our TCDM interconnect protocol. I intend to add a reference here at some point too.

Reply:
It would be really nice to have some sort of page with all the documentation links to different projects.